File: python-ai-service/plaid_client.py
import os
import plaid
from plaid.api import plaid_api

_client: plaid_api.PlaidApi | None = None

_ENV_MAP = {
    "sandbox": plaid.Environment.Sandbox,
    "production": plaid.Environment.Production,
}


def get_plaid_client() -> plaid_api.PlaidApi:
    global _client
    if _client is not None:
        return _client

    env_name = os.getenv("PLAID_ENV", "sandbox").lower()
    host = _ENV_MAP.get(env_name, plaid.Environment.Sandbox)

    configuration = plaid.Configuration(
        host=host,
        api_key={
            "clientId": os.getenv("PLAID_CLIENT_ID", ""),
            "secret": os.getenv("PLAID_SECRET", ""),
        },
    )
    api_client = plaid.ApiClient(configuration)
    _client = plaid_api.PlaidApi(api_client)
    return _client

File: src/lib/server/plaidClient.ts
/**
 * Singleton Plaid API client.
 * Reads PLAID_CLIENT_ID, PLAID_SECRET, and PLAID_ENV from SvelteKit private env.
 * PLAID_ENV accepts: 'sandbox' | 'production' (defaults to 'sandbox')
 */

import { Configuration, PlaidApi, PlaidEnvironments } from 'plaid';
import { env } from '$env/dynamic/private';

let _client: PlaidApi | null = null;

export function getPlaidClient(): PlaidApi {
	if (_client) return _client;

	const plaidEnv = env.PLAID_ENV === 'production'
		? PlaidEnvironments.production
		: PlaidEnvironments.sandbox;

	const config = new Configuration({
		basePath: plaidEnv,
		baseOptions: {
			headers: {
				'PLAID-CLIENT-ID': env.PLAID_CLIENT_ID ?? '',
				'PLAID-SECRET': env.PLAID_SECRET ?? ''
			}
		}
	});

	_client = new PlaidApi(config);
	return _client;
}
